diff -rNu GraphicsMagick-1.3.17.orig/coders/Makefile.am GraphicsMagick-1.3.17/coders/Makefile.am
--- GraphicsMagick-1.3.17.orig/coders/Makefile.am	2012-10-14 05:30:21.000000000 +0800
+++ GraphicsMagick-1.3.17/coders/Makefile.am	2013-01-06 17:19:47.128777091 +0800
@@ -56,6 +56,11 @@
 MAGICK_X11_CODER_SRCS = coders/x.c coders/xwd.c
 endif
 
+if HasWEBP
+MAGICK_WEBP_MODULES = coders/webp.la
+MAGICK_WEBP_SRCS = coders/webp.c
+endif
+
 MAGICK_CODER_SRCS = \
 	$(MAGICK_PNG_SRCS) \
 	coders/art.c \
@@ -152,7 +157,8 @@
 	$(MAGICK_JPEG_SRCS) \
 	$(MAGICK_JP2_SRCS) \
 	$(MAGICK_TIFF_SRCS) \
-	$(MAGICK_X11_CODER_SRCS)
+	$(MAGICK_X11_CODER_SRCS) \
+	$(MAGICK_WEBP_SRCS)
 
 if WITH_MODULES
 coders_LTLIBRARIES = \
@@ -245,7 +251,8 @@
 	coders/yuv.la \
 	$(MAGICK_DPS_MODULES) $(MAGICK_FPX_MODULES) $(MAGICK_GDI32_MODULES) \
 	$(MAGICK_JBIG_MODULES) $(MAGICK_JPEG_MODULES) $(MAGICK_JP2_MODULES) \
-	$(MAGICK_PNG_MODULES) $(MAGICK_TIFF_MODULES) $(MAGICK_X11_CODER_MODULES)
+	$(MAGICK_PNG_MODULES) $(MAGICK_TIFF_MODULES) $(MAGICK_X11_CODER_MODULES) \
+	$(MAGICK_WEBP_MODULES)
 else
 coders_LTLIBRARIES =
 endif # WITH_MODULES
@@ -849,3 +856,9 @@
 coders_yuv_la_CPPFLAGS     = $(MODULE_EXTRA_CPPFLAGS)
 coders_yuv_la_LDFLAGS      = $(MODULECOMMONFLAGS)
 coders_yuv_la_LIBADD       = $(LIBMAGICK)
+
+# WEBP coder module
+coders_webp_la_SOURCES     = coders/webp.c
+coders_webp_la_CPPFLAGS    = $(MODULE_EXTRA_CPPFLAGS)
+coders_webp_la_LDFLAGS     = $(MODULECOMMONFLAGS)
+coders_webp_la_LIBADD      = $(LIBMAGICK) $(LIB_WEBP)
diff -rNu GraphicsMagick-1.3.17.orig/configure.ac GraphicsMagick-1.3.17/configure.ac
--- GraphicsMagick-1.3.17.orig/configure.ac	2012-10-14 05:30:21.000000000 +0800
+++ GraphicsMagick-1.3.17/configure.ac	2013-01-06 17:19:47.129777092 +0800
@@ -662,6 +662,16 @@
     DISTCHECK_CONFIG_FLAGS="${DISTCHECK_CONFIG_FLAGS} --with-jbig=$with_jbig "
 fi
 
+# Disable WEBP.
+AC_ARG_WITH(webp,
+        AS_HELP_STRING([--without-webp],
+                             [disable WEBP support]),
+        [with_webp=$withval],
+        [with_webp='yes'])
+if test "$with_webp" != 'yes' ; then
+    DISTCHECK_CONFIG_FLAGS="${DISTCHECK_CONFIG_FLAGS} --with-webp=$with_webp "
+fi
+
 # Disable JPEG.
 AC_ARG_WITH(jpeg,
 	      AS_HELP_STRING([--without-jpeg],
@@ -2356,6 +2366,40 @@
 AC_SUBST(LIB_JBIG)
 
 #
+# Check for WEBP
+#
+have_webp='no'
+LIB_WEBP=''
+if test "$with_webp" != 'no'
+then
+    AC_MSG_CHECKING(for WEBP support )
+    AC_MSG_RESULT()
+    failed=0
+    passed=0
+    AC_CHECK_HEADER(webp/decode.h,passed=`expr $passed + 1`,failed=`expr $failed + 1`,)
+    AC_CHECK_LIB(webp,WebPDecodeRGB,passed=`expr $passed + 1`,failed=`expr $failed + 1`,)
+    AC_MSG_CHECKING(if WEBP package is complete)
+    if test $passed -gt 0
+    then
+    if test $failed -gt 0
+    then
+  AC_MSG_RESULT(no -- some components failed test)
+        have_webp='no (failed tests)'
+    else
+  LIB_WEBP='-lwebp'
+  LIBS="$LIB_WEBP $LIBS"
+  AC_DEFINE(HasWEBP,1,Define if you have WEBP library)
+  AC_MSG_RESULT(yes)
+        have_webp='yes'
+    fi
+    else
+        AC_MSG_RESULT(no)
+    fi
+fi
+AM_CONDITIONAL(HasWEBP, test "$have_webp" = 'yes')
+AC_SUBST(LIB_WEBP)
+
+#
 # Check for XML
 #
 have_xml='no'
@@ -3153,6 +3197,10 @@
    DELEGATES="$DELEGATES jbig"
    MAGICK_FEATURES="$MAGICK_FEATURES JBIG"
 fi
+if test "$have_webp"   = 'yes' ; then
+   DELEGATES="$DELEGATES webp"
+   MAGICK_FEATURES="$MAGICK_FEATURES WEBP"
+fi
 if test "$have_png$have_jpeg" = 'yesyes' ; then
    DELEGATES="$DELEGATES jng"
    MAGICK_FEATURES="$MAGICK_FEATURES JNG"
@@ -3305,7 +3353,7 @@
 then
   MAGICK_DEP_LIBS="$LIB_LCMS $LIB_TTF $LIB_GS $LIB_XEXT $LIB_IPC $LIB_X11 $LIB_LZMA $LIB_BZLIB $LIB_ZLIB $LIB_TRIO $LIB_GDI32 $LIB_MATH $LIB_OMP $LIB_UMEM $LIB_THREAD"
 else
-  MAGICK_DEP_LIBS="$LIB_JBIG $LIB_LCMS $LIB_TIFF $LIB_TTF $LIB_JP2 $LIB_JPEG $LIB_GS $LIB_PNG $LIB_FPX $LIB_WMF $LIB_DPS $LIB_XEXT $LIB_IPC $LIB_X11 $LIB_LZMA $LIB_BZLIB $LIB_XML $LIB_ZLIB $LIB_TRIO $LIB_GDI32 $LIB_MATH $LIB_OMP $LIB_UMEM $LIB_THREAD"
+  MAGICK_DEP_LIBS="$LIB_JBIG $LIB_WEBP $LIB_LCMS $LIB_TIFF $LIB_TTF $LIB_JP2 $LIB_JPEG $LIB_GS $LIB_PNG $LIB_FPX $LIB_WMF $LIB_DPS $LIB_XEXT $LIB_IPC $LIB_X11 $LIB_LZMA $LIB_BZLIB $LIB_XML $LIB_ZLIB $LIB_TRIO $LIB_GDI32 $LIB_MATH $LIB_OMP $LIB_UMEM $LIB_THREAD"
 fi
 AC_SUBST(MAGICK_DEP_LIBS)
 
@@ -3427,6 +3475,7 @@
 echo "Ghostscript fonts --with-gs-font-dir=$with_gs_font_dir    $result_ghostscript_font_dir"
 echo "Ghostscript lib   --with-gslib=$with_gslib       	$have_gslib"
 echo "JBIG              --with-jbig=$with_jbig        	$have_jbig"
+echo "WEBP              --with-webp=$with_webp        	$have_webp"
 echo "JPEG v1           --with-jpeg=$with_jpeg        	$have_jpeg"
 echo "JPEG-2000         --with-jp2=$with_jp2          	$have_jp2"
 echo "LCMS v1           --with-lcms=$with_lcms        	$have_lcms"
diff -rNu GraphicsMagick-1.3.17.orig/magick/magic.c GraphicsMagick-1.3.17/magick/magic.c
--- GraphicsMagick-1.3.17.orig/magick/magic.c	2011-12-25 03:34:17.000000000 +0800
+++ GraphicsMagick-1.3.17/magick/magic.c	2013-01-06 17:19:47.129777092 +0800
@@ -59,6 +59,7 @@
 StaticMagic[] =
 {
 #define MAGIC(name,offset,magic) {name,(unsigned char *)magic,sizeof(magic)-1,offset}
+  MAGIC("WEBP", 8, "WEBP"),
   MAGIC("AVI", 0, "RIFF"),
   MAGIC("8BIMWTEXT", 0, "8\000B\000I\000M\000#"),
   MAGIC("8BIMTEXT", 0, "8BIM#"),
diff -rNu GraphicsMagick-1.3.17.orig/magick/static.c GraphicsMagick-1.3.17/magick/static.c
--- GraphicsMagick-1.3.17.orig/magick/static.c	2012-10-14 05:30:22.000000000 +0800
+++ GraphicsMagick-1.3.17/magick/static.c	2013-01-06 17:19:47.129777092 +0800
@@ -246,6 +246,9 @@
   RegisterVIDImage();
   RegisterVIFFImage();
   RegisterWBMPImage();
+#if defined(HasWEBP)
+  RegisterWEBPImage();
+#endif
   RegisterWMFImage();
   RegisterWPGImage();
 #if defined(HasX11)
@@ -401,6 +404,9 @@
   UnregisterVIDImage();
   UnregisterVIFFImage();
   UnregisterWBMPImage();
+#if defined(HasWEBP)
+  UnregisterWEBPImage();
+#endif
   UnregisterWMFImage();
   UnregisterWPGImage();
 #if defined(HasX11)
diff -rNu GraphicsMagick-1.3.17.orig/magick/static.h GraphicsMagick-1.3.17/magick/static.h
--- GraphicsMagick-1.3.17.orig/magick/static.h	2012-10-14 05:30:22.000000000 +0800
+++ GraphicsMagick-1.3.17/magick/static.h	2013-01-06 17:19:47.129777092 +0800
@@ -142,6 +142,7 @@
   RegisterVIFFImage(void),
   RegisterVSTImage(void),
   RegisterWBMPImage(void),
+  RegisterWEBPImage(void),
   RegisterWMFImage(void),
   RegisterWPGImage(void),
   RegisterXImage(void),
@@ -280,6 +281,7 @@
   UnregisterVIFFImage(void),
   UnregisterVSTImage(void),
   UnregisterWBMPImage(void),
+  UnregisterWEBPImage(void),
   UnregisterWMFImage(void),
   UnregisterWPGImage(void),
   UnregisterXImage(void),
